AppEngine: mounted volume permission error if container application is not running as root

### Summary:

For an application to have write access to a mounted volume, the application's user context must have appropriate access.  If the application is not running as root, then it will often be the case that the permissions will not match.  

NOTE: it is recommended that developers should NOT run their containerized applications as root.  Therefore, more and more applications should be moving in this direction.

Example:
Installation of NodeRED runs into this error (full error can be found at the bottom of this ticket):
```
Error: EACCES: permission denied, copyfile '/usr/src/node-red/node_modules/node-red/settings.js' -> '/data/settings.js'
```

Kubernetes provides a `securityContext.fsGroup` property that tells Kubernetes to update the permissions on any attached volumes to grant permissions to a specific group, and ensures the running context is part of this group.

**Potential solution**:
1. We could expose a parameter through the AppEngine spec that allows a developer to control this `fsGroup` (or maybe even the whole security context).   However, I think this might be a very common need, and if the intention is for AppEngine to simplify development, it may be easier just to have a global default.
2. Blanket add an fsGroup to all AppEngine deployments.
```
securityContext: {
  fsGroup: 1000   # or any number
}
```

### Failure Logs

Full NodeRED error:
```
> node-red-docker@1.2.7 start /usr/src/node-red
> node $NODE_OPTIONS node_modules/node-red/red.js $FLOWS "--userDir" "/data"
fs.js:114
    throw err;
    ^
Error: EACCES: permission denied, copyfile '/usr/src/node-red/node_modules/node-red/settings.js' -> '/data/settings.js'
    at Object.copyFileSync (fs.js:1728:3)
    at copyFile (/usr/src/node-red/node_modules/fs-extra/lib/copy-sync/copy-sync.js:68:8)
    at onFile (/usr/src/node-red/node_modules/fs-extra/lib/copy-sync/copy-sync.js:53:25)
    at getStats (/usr/src/node-red/node_modules/fs-extra/lib/copy-sync/copy-sync.js:48:44)
    at startCopy (/usr/src/node-red/node_modules/fs-extra/lib/copy-sync/copy-sync.js:38:10)
    at handleFilterAndCopy (/usr/src/node-red/node_modules/fs-extra/lib/copy-sync/copy-sync.js:33:10)
    at Object.copySync (/usr/src/node-red/node_modules/fs-extra/lib/copy-sync/copy-sync.js:26:10)
    at Object.<anonymous> (/usr/src/node-red/node_modules/node-red/red.js:125:20)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-red-docker@1.2.7 start: `node $NODE_OPTIONS node_modules/node-red/red.js $FLOWS "--userDir" "/data"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-red-docker@1.2.7 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

## Checklist
- [ ] I am running the latest version
- [ ] I checked the documentation and found no answer
- [ ] I checked to make sure that this issue has not already been filed
- [ ] I'm reporting the issue to the correct repository (for multi-repository projects)
- [ ] I have provided sufficient information for the team

--- a/packages/appengine/src/templates/deployment.ts
+++ b/packages/appengine/src/templates/deployment.ts
@@ -38,6 +38,10 @@ export function getDeploymentTemplate(
                     labels
                 },
                 spec: {
+                    securityContext: {
+                        // See https://github.com/c6o/provisioners/issues/182
+                        fsGroup: 1000,
+                    },
                     containers: [{
                         name,
                         image,

